### PR TITLE
Add MultiDataProvider

### PR DESCRIPTION
**Pull request by prin-r**
[see specification](https://github.com/open-web3-stack/open-runtime-module-library/issues/245)

Runtime

- ✅ AcalaDataProvider: orml_oracle::Instance1
- ✅ BandDataProvider: orml_oracle::Instance2
- ✅ enum DataProviderId
   - AcalaDataProvier
   - BandDataProvider
   - AggergatedDataProvider
- ✅ create_midean_value_data_provider(AggregatedDataProvider, AcalaDataProvider, BandDataProvider)
- ✅ struct MultiDataProviderImpl
- ✅ impl MultiDataProvider<DataProviderId, CurrencyId, Price>
- ✅ Prices::Source = AggregatedDataProvider<Two, OneHour, CurrencyId, Price>

### CODE REVIEW

File: runtime/dev/src/lib.rs [hunk 1]
```diff
@@ -658,18 +673,44 @@ parameter_types! {
...
+	type OracleValue = Price;
+	type UnsignedPriority = OracleUnsignedPriority;
+	type AuthorityId = orml_oracle::AuthorityId;
+}
+
+pub type TimeStampedPrice = TimestampedValue<Price, Moment>;
+
+create_median_value_data_provider!(AggregatedDataProviderImpl, AcalaOracle, BandOracle);
+
+struct MultiDataProviderImpl;
```
Comment (shaopengw) [inline runtime/dev/src/lib.rs:704]:
```suggestion
struct DataProviders;
```

File: primitives/src/lib.rs [hunk 2]
```diff
@@ -91,3 +91,11 @@ pub enum AuthoritysOriginId {
...
 	DSWF,
 }
+
+#[derive(Encode, Decode, Eq, PartialEq, Copy, Clone, RuntimeDebug, PartialOrd, Ord)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub enum DataProviderId {
+	AcalaDataProvier = 0,
+	BandDataProvider = 1,
+	AggergatedDataProvider = 2,
+}
```
Comment (shaopengw) [inline primitives/src/lib.rs:101]:
```suggestion
pub enum DataProviderId {
	Acala = 0,
	Band = 1,
	Aggergated = 2,
}
```

File: runtime/dev/src/lib.rs [hunk 3]
```diff
@@ -658,18 +673,44 @@ parameter_types! {
...
+	type Time = Timestamp;
+	type OracleKey = CurrencyId;
+	type OracleValue = Price;
+	type UnsignedPriority = OracleUnsignedPriority;
+	type AuthorityId = orml_oracle::AuthorityId;
+}
+
+pub type TimeStampedPrice = TimestampedValue<Price, Moment>;
+
+create_median_value_data_provider!(AggregatedDataProviderImpl, AcalaOracle, BandOracle);
```
Comment (shaopengw) [inline runtime/dev/src/lib.rs:702]:
```suggestion
create_median_value_data_provider!(AggregatedDataProvider, AcalaOracle, BandOracle);
```

**Comment by shaopengw** [comment]
We might need some integration tests to cover changes, but I'm not sure if in the scope of this PR. cc @xlc @wangjj9219

**Comment by wangjj9219** [comment]
> We might need some integration tests to cover changes, but I'm not sure if in the scope of this PR. cc @xlc @wangjj9219

Currently ours services only use acala provider. Can merge this PR first, and then I will write some integration tests to cover the cases of multiple oracle.

I have another question.  Different oracles will be governed by their respective operators in the future? 
Unlimited members number and refund tx fee for valid `feed_value` tx maybe cause some problems.

**Comment by xlc** [comment]
We will work out the exact governance later. But Acala Council should always have ability to remove bad oracle operator.

**Comment by prin-r** [comment]
@xlc , @shaopengw 

As discussed on discord, I'm handing this off to you guys. Please take this forward by adding tests. Let me know if I need to do anything.

File: primitives/src/lib.rs [hunk 4]
```diff
@@ -91,3 +91,11 @@ pub enum AuthoritysOriginId {
 	HomaTreasury,
 	DSWF,
 }
+
+#[derive(Encode, Decode, Eq, PartialEq, Copy, Clone, RuntimeDebug, PartialOrd, Ord)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub enum DataProviderId {
+	Acala = 0,
+	Band = 1,
+	Aggregated = 2,
```
Comment (xlc) [inline primitives/src/lib.rs:100]: Put `Aggregated = 0` and make it the default

File: runtime/dev/src/lib.rs [hunk 5]
```diff
@@ -659,17 +671,44 @@ parameter_types! {
...
+	AggregatedDataProvider,
+	CurrencyId,
+	Price,
+	TimeStampedPrice,
+	[AcalaOracle, BandOracle]
+);
+// Aggregated data provider cannot feed.
+impl DataFeeder<CurrencyId, Price, AccountId> for AggregatedDataProvider {
+	fn feed_value(_: AccountId, _: CurrencyId, _: Price) -> DispatchResult {
+		Ok(())
```
Comment (xlc) [inline runtime/dev/src/lib.rs:708]:
```suggestion
		Err("Not supported".into())
```

**Review by xlc — APPROVED** [review]

**Review by shaopengw — APPROVED** [review]

**Review by xlc — APPROVED** [review]

**Review by wangjj9219 — APPROVED** [review]

**Pull request merged**